### PR TITLE
Stripe 2.8 support

### LINKED ISSUE #410: Stripe 2.0.0 released, updated dependencies
Stripe released version 2.0.0 of the gem recently with the following changes:

* Drop support for Ruby 1.9
* Allow HTTP client that makes Stripe calls to be configured via Faraday
* Drop RestClient
* Switch to OpenAPI 2.0 spec and generated fixtures in test suite
* Switch to Webmock in test suite

`stripe-ruby-mock` has us locked into `~> 1.31`. Any plans to update that dependency? If the 1.9 support is a concern maybe fork the current version of this gem and have it retain 1.9 support while this one moves forward with the `stripe` gem itself?

### PR DESCRIPTION
**Pull request by Borzik**
This PR fixes all references to methods that were removed (or updated) from Stripe v2.0.0 until v2.8.0.
I've tried to avoid any unnecessary changes (except of some trailing spaces that were removed by the editor automatically).
Public API hasn't been changed as well as configuration.

UPD: I'd appreciate if someone who is familiar with the project could help me decide what to do with deprecated recipients. I've disabled some specs because they are no longer passing (you can see the same issue in #423). Fixes #410

### CODE REVIEW

**Comment by benoror** [comment]
@gilbert @alexmamonchik any plans to ship this soon? We need to upgrade to latests stripe gem version, and this is a blocker for us. Any info would be appreciated, regards!

**Comment by nlively** [comment]
@Borzik, I tried using this branch to mock Stripe's SubscriptionItem API, and had some issues.  Code is here: https://github.com/boomr/stripe-ruby-mock.  I have 1 additional commit since this PR, but I don't think it's merge-worthy.  Would appreciate your insights on this to make sure i'm following existing patterns with this gem and to help round out on SubscriptionItems.

File: .travis.yml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 language: ruby
 rvm:
-  - 1.9.3
-  - 2.0.0
-  - 2.1.6
+  - 2.1.10
```
Comment (alexmamonchik) [inline .travis.yml]: actually, I think we still need to support ruby 2.0.

File: stripe-ruby-mock.gemspec [hunk 2]
```diff
@@ -17,7 +17,7 @@ Gem::Specification.new do |gem|
   gem.test_files    = gem.files.grep(%r{^(test|spec|features)/})
   gem.require_paths = ['lib']
 
-  gem.add_dependency 'stripe', '~> 1.31'
+  gem.add_dependency 'stripe', '~> 2.8'
```
Comment (alexmamonchik) [inline stripe-ruby-mock.gemspec]: also, I think minimum required stripe gem version is 2.0.0, not 2.8.

File: .travis.yml (hunk 1 above)
Comment (Borzik) [inline .travis.yml]: Yes, I've been thinking of adding ruby 2.0.x there too, but since we've got only three api keys, I've decided to go with ruby 2.1 and higher. Could you please give me a test key to use with ruby 2.0.x? We also have 2.4 already, so probably we'll need to have it as well; then I need two more test api keys.

Comment (alexmamonchik) [inline .travis.yml]: try those keys, but I'm not sure https://github.com/rebelidealist/stripe-ruby-mock/blob/updated-stripe-keys/.travis.yml#L16

Comment (Borzik) [inline .travis.yml]: Yes, I already use them for rubies 2.1, 2.2 and 2.3. If it's ok to use them to test against two ruby versions simultaneously, I can do it, although I'm not sure it won't create any overlaps. I've taken these from your PR to have the latest Stripe API version for testing.

Comment (alexmamonchik) [inline .travis.yml]: Actually, we have key for ruby 2.0 - you just removed it :-) https://github.com/rebelidealist/stripe-ruby-mock/blob/master/.travis.yml#L16

Comment (Borzik) [inline .travis.yml]: yes, I know :) I just don't know which Stripe API version is attached to those keys. E.g. Recipients feature may work on some older APIs, but will not work now, so I want to make sure it will test against an up-to-date API version.

File: stripe-ruby-mock.gemspec (hunk 2 above)
Comment (Borzik) [inline stripe-ruby-mock.gemspec]: I've ran test suite against `2.0`, but it had some specs failed, and it worked fine with `2.1` only. I guess there were some fixes in 2.1. 
There's not such a big difference between 2.0 and 2.1, so I think it will worth upgrading to 2.1 or even to the most recent version, but 2.1 should work since tests don't fail.

Comment (alexmamonchik) [inline stripe-ruby-mock.gemspec]: existing code works fine with stripe 1.58. With 2.0 it totally broken. The main problem in ```ruby
Stripe::StripeClient.active_client.method(:request)
```
you fixed it. I want to make a release with 1.31-1.58 stripe gem for old projects and 2.0 and newest for new projects. You set 2.1 as minimum and it means that people who use 2.0 stripe can't use any of stripe-ruby-mock release.

Comment (alexmamonchik) [inline stripe-ruby-mock.gemspec]: what's wrong with 2.0? Let's fix it and merge this PR

Comment (Borzik) [inline stripe-ruby-mock.gemspec]:
``` 
=== 2.0.3 2017-03-16
* Fix marshalling of `StripeObjects` that have an embedded client
```
I've tested it with 2.0.3 and tests didn't fail, but failed with 2.0.2. So I think that's the issue that causes some specs to fail. I've lowered required version to 2.0.3

Comment (alexmamonchik) [inline stripe-ruby-mock.gemspec]: perfect! Thanks

**Comment by alexmamonchik** [comment]
@Borzik please, pull master - it has some travis updates

**Comment by alexmamonchik** [comment]
@Borzil we are on the last step before merge. Please, pull master one more time

**Comment by Borzik** [comment]
@alexmamonchik 
Done

**Comment by mcmire** [comment]
@Borzik Just wanted to clarify - if I understand correctly, this PR fixes any incompatibilities/deprecations with Stripe 2.8 but doesn't necessary add anything that versions 2.0 through 2.8 have added?

**Comment by alexmamonchik** [comment]
@mcmire also, updated stripe keys for live tests, locked minimum stripe gem requirements

**Comment by Borzik** [comment]
@mcmire yes, this was intended to fix incompatibilities with stripe versions 2.0.0 to 2.8, it doesn't add any new features.

**Comment by alexmamonchik** [comment]
follow #442 PR. from this we'll use newest API and stripe versions

**Comment by mcmire** [comment]
Awesome sauce. Thanks for doing this by the way!
